### PR TITLE
Requesting a pull to mozilla:master from k88hudson:t2082

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

File: src/dialog/dialogs/first-run.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+If a copy of the MIT license was not distributed with this file, you can
+obtain one at http://www.mozillapopcorn.org/butter-license.txt -->
+
+<!-- This is just a placeholder, really -->
+<div class="butter-first-run-dialog butter-feedback-form butter-dialog">
+  <div class="butter-logo"></div>
+  <div class="butter-dialog-offset">
+    <div class="butter-dialog-title">Welcome to Popcorn Maker Beta</div>
+    <div class="butter-dialog-body">
+<p>This version of Popcorn Maker is not yet completed, so please let us know via the feedback form if you have errors.</p>
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.html]: spppppppaaaaaaaaaceeeeeeeeee

File: src/dialog/dialogs/first-run.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<p>Work you create here may expire - we'll launch at the upcoming Mozilla Festival on November 9th!</p>
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.html]: ^

File: src/dialog/dialogs/first-run.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        editor.removeChild( overlayDiv );
+        dialog.activity( "default-close" );
+      }
+   
+      function secondStep() {
+        closeFirstRunDialog();
+         ToolTip.create({
+          name: "tooltip-popup",
+          element: popupTile,
+          message: '<h3>Event</h3>Try dragging this to the stage',
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.js]: Double Quotes.

File: src/dialog/dialogs/first-run.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+          hidden: false
+        });
+        ToolTip.create({
+          name: "tooltip-media",
+          element: mediaEditorButton,
+          top: "60px",
+          message: '<h3>Media Editor</h3>Change your media source here!<span class="center-div"><span class="media-icon youtube-icon"></span><span class="media-icon vimeo-icon"></span><span class="media-icon soundcloud-icon"></span><span class="media-icon html5-icon"></span></span>',
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.js]: Double quotes.

Comment (mjschranz) [inline src/dialog/dialogs/first-run.js]: Can you multiline this string too? Easier to read.

File: src/dialog/dialogs/first-run.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        popupTile.removeEventListener( "mouseover", secondStep, false );
+       }
+
+      popupTile.addEventListener( "mouseover", secondStep, false );
+
+      window.addEventListener( "click", secondStep, false );
+
+
+
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.js]: Remove two of these new lines.

File: src/dialog/dialogs/first-run.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      var EVENT_EDITOR_NAME = "butter-first-run-editor-img",
+          MEDIA_EDITOR_NAME = "butter-first-run-media-img";
+
+      var editor = document.querySelector( ".butter-editor-area" ),
+          mediaEditorButton = document.querySelector( ".butter-editor-header-media" ),
+          popupTile = document.querySelector( ".butter-plugin-tile[data-popcorn-plugin-type=popup]" ),
+          overlayDiv = document.createElement( "div" ),
+          popupTooltip,
+          mediaTooltip;
```
Comment (mjschranz) [inline src/dialog/dialogs/first-run.js:19]: Why two var lists? If one is a set of CONSTS then put them before the Dialog.register.

File: src/dialog/dialogs/first-run.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        mediaEditorButton.classList.remove( "overlay-highlight" );
+        popupTile.classList.remove( "overlay-highlight" );
+        editor.removeChild( overlayDiv );
+        document.body.classList.remove( "first-run" );
+        dialog.activity( "default-close" );
+      }
+   
+      function secondStep() {
+        closeFirstRunDialog();
+         popupTooltip = ToolTip.create({
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Indentation.

File: src/dialog/dialogs/first-run.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        });
+
+        setTimeout( function() {
+          if ( mediaTooltip ) {
+            mediaTooltip.parentNode.removeChild( mediaTooltip );
+          }
+          if ( popupTooltip ) {
+            popupTooltip.parentNode.removeChild( popupTooltip );
+          }
+        }, 3000 );
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Const.

File: src/dialog/dialogs/first-run.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        popupTile.removeEventListener( "mouseover", secondStep, false );
+       }
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: This whole block needs better indentation.

File: src/first-run.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+ */
+define( [ "dialog/dialog", "util/cookie" ], function( Dialog, Cookie ) {
+  return {
+    init: function( config ) {
+
+      var _dialog;
+      
+      if ( !Cookie.isPopcornCookieSet() || window.location.search.match( "alwaysFirst" ) ) {
+        Cookie.setPopcornCookie();
+        _dialog = Dialog.spawn( "first-run", {data:"foo"} );
```
Comment (secretrobotron) [inline src/first-run.js]: Spacing around `{`.

File: src/dialog/dialogs/first-run.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+      overlayDiv.classList.add( "butter-modal-overlay-dark-bg" );
+      overlayDiv.classList.add( "fade-in" );
+
+      editor.appendChild( overlayDiv );
+      document.body.classList.add( "first-run" );
+
+      popupTile.classList.add( "overlay-highlight" );
+
+      function closeFirstRunDialog() {
+        window.removeEventListener( "click", secondStep );
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Please put `false` as the last param here.

File: src/dialog/dialogs/first-run.js [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+          name: "tooltip-popup",
+          element: popupTile,
+          message: '<h3>Event</h3>Try dragging this to the stage',
+          hidden: false
+        });
+        mediaTooltip = ToolTip.create({
+          name: "tooltip-media",
+          element: mediaEditorButton,
+          top: "60px",
+          message: '<h3>Media Editor</h3>Change your media source here!<span class="center-div"><span class="media-icon youtube-icon"></span><span class="media-icon vimeo-icon"></span><span class="media-icon soundcloud-icon"></span><span class="media-icon html5-icon"></span></span>',
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Use double quotes.

File: src/dialog/dialogs/first-run.js [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+        document.body.classList.remove( "first-run" );
+        dialog.activity( "default-close" );
+      }
+   
+      function secondStep() {
+        closeFirstRunDialog();
+        popupTooltip = ToolTip.create({
+          name: "tooltip-popup",
+          element: popupTile,
+          message: '<h3>Event</h3>Try dragging this to the stage',
```
Comment (secretrobotron) [inline src/dialog/dialogs/first-run.js]: Use double quotes.

**Pull request merged**